### PR TITLE
refine username validation to not be able to use an email as username

### PR DESCRIPTION
**Pull request by Mat-moran**
An user can use an email as usernam. This can conflict when using method that takes an argument as username or password and find the user for any of them. 

## Test Plan

I have verify locally that the regex is OK using an username with `@` gives a zod error and if the username does not contain `@` it works as expected.

### CODE REVIEW

File: app/utils/user-validation.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ export const usernameSchema = z
 	.string()
 	.min(3, { message: 'Username is too short' })
 	.max(20, { message: 'Username is too long' })
+	.regex(new RegExp("^[^@]*$"), "username can not contains @")
```
Comment (kentcdodds) [inline app/utils/user-validation.ts:7]: Wouldn't this work?

```suggestion
	.regex(/@/, 'Username can not contain @')
```

Comment (Mat-moran) [reply app/utils/user-validation.ts:7]: does not work, we can avoid using `new RegExp(regular expression)` but only `/@/` is telling zod that the username should have the symbol `@`

File: app/utils/user-validation.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@ export const usernameSchema = z
 	.string()
 	.min(3, { message: 'Username is too short' })
 	.max(20, { message: 'Username is too long' })
+	.regex(/^[^@]*$/, "username can not contains @")
```
Comment (kentcdodds) [inline app/utils/user-validation.ts:7]:
```suggestion
	.regex(/^[^@]*$/, 'Username can not contain @')
```

File: app/utils/user-validation.ts [hunk 3]
```diff
@@ -4,6 +4,7 @@ export const usernameSchema = z
 	.string()
 	.min(3, { message: 'Username is too short' })
 	.max(20, { message: 'Username is too long' })
+	.regex(/^[^@]*$/, 'Username can not contain @')
```
Comment (kentcdodds) [inline app/utils/user-validation.ts:7]:
```suggestion
	.regex(/^[^@]*$/, 'Username can not contain "@"')
```

**Review by kentcdodds — APPROVED** [review]
Thanks!

**Pull request merged**